app/services/sso/openid_connect.rb

```diff
@@ -56,11 +56,14 @@ def update_session(payload)
     end
 
     def find_or_create_user_from_jwt(payload)
+      attrs = { login: payload['preferred_username'],
+                mail: payload['email'],
+                firstname: payload['given_name'],
+                lastname: payload['family_name'],
+              }
+      attrs[:groups] = payload['groups'].split(',') if payload['groups'].present?
```

Comment:
I'm not sure what happened to my comment here, perhaps I forgot to save it. Looks like according to the [keycloak docs](https://www.keycloak.org/docs/latest/securing_apps/#group-claims), the groups claim is sent as an array, not as a string.

Reply:
I was doing this via a[ different way](https://www.keycloak.org/docs/7.0/server_admin/#_protocol-mappers) but the docs pointed out by you is a better way of doing it :) Will update the PR asap.

Reply:
To achieve the [Group claim](https://www.keycloak.org/docs/latest/securing_apps/#group-claims) as per the documentation, one has to follow these steps on the keycloak side. https://stackoverflow.com/revisions/56372674/2